### PR TITLE
[Feature:TAGrading] Peer graders student anonymization

### PR DESCRIPTION
**Pull request by jchulton**
### Please check if the PR fulfills these requirements:

* [ ] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
A student peer grader can see the student ids they are assigned to grade, both on the details page and via inspect element

### What is the new behavior?
A student peer grader can only see the anon_id of the students they are assigned to peer grade

### CODE REVIEW

File: site/app/controllers/grading/ElectronicGraderController.php [hunk 1]
```diff
@@ -926,18 +937,23 @@ public function showGrading($gradeable_id, $who_id = '', $from = "", $to = null,
             // Reassign who_id
             if (!is_null($goToStudent)) {
                 $who_id = $goToStudent->getId();
+                if ($peer && !$team) {
+                    $who_id = $goToStudent->getAnonId();
+                }
```
Comment (emaicus) [inline site/app/controllers/grading/ElectronicGraderController.php:942]:
```
$who_id = $peer && !$team ? $goToStudent->getAnonId() :   $goToStudent->getId();
```

File: site/app/controllers/grading/ElectronicGraderController.php [hunk 2]
```diff
@@ -926,18 +937,23 @@ public function showGrading($gradeable_id, $who_id = '', $from = "", $to = null,
...
 
         // Get the graded gradeable for the submitter we are requesting
-        $graded_gradeable = $this->tryGetGradedGradeable($gradeable, $who_id, false);
+        $graded_gradeable = null;
+        if ($peer && !$team) {
+            $graded_gradeable = $this->tryGetGradedGradeable($gradeable, $this->core->getQueries()->getUserFromAnon($who_id)[$who_id], false);
+        }
+        else {
+            $graded_gradeable = $this->tryGetGradedGradeable($gradeable, $who_id, false);
+        }
```
Comment (emaicus) [inline site/app/controllers/grading/ElectronicGraderController.php:953]: As above, can we simplify this to something like:
```
$my_id = $peer && !$team ? $this->core->getQueries()->getUserFromAnon($who_id)[$who_id] : $who_id;
$graded_gradeable = $this->tryGetGradedGradeable($gradeable, $my_id, false);
```

**Comment by emaicus** [comment]
Found two places where a user_id still appears in the html of the grading page.
```
<button class="btn btn-default key_to_click" tabindex="0" onclick="downloadSubmissionZip('yes','aufded', 0)">Download Zip File</button>
```
and
```
    function loadFile(name, path, page_num = 0) {
        let extension = name.split('.').pop();
        if (extension == "pdf") {
            $('#pdf_annotation_bar').show();
            $('#save_status').show();
            return $.ajax({
                type: 'POST',
                url: buildCourseUrl(['gradeable', 'yes', 'grading', 'pdf']),
                data: {
                    'user_id': 'aufded',
                    'filename': name,
                    'file_path': path,
                    'page_num': page_num,
                    'csrf_token': csrfToken
                },
                success: function(data){
                    $('#file_content').append(data);
                }
            });
        }
        else {
            $('#save_status').hide();
            $('#file_content').append("<div id=\"file_viewer_full_panel\" class=\"full_panel\" data-file_name=\"\" data-file_url=\"\"></div>");
            $("#file_viewer_full_panel").empty();
            $("#file_viewer_full_panel").attr("data-file_name", "");
            $("#file_viewer_full_panel").attr("data-file_url", "");
            $("#file_viewer_full_panel").attr("data-file_name", name);
            $("#file_viewer_full_panel").attr("data-file_url", path);
            openFrame(name, path, "full_panel");
            $("#file_viewer_full_panel_iframe").height("100%");
        }
    }

```

**Comment by emaicus** [comment]
The user's name also appears in the overall comments box.

File: site/app/controllers/grading/ElectronicGraderController.php [hunk 3]
```diff
@@ -892,7 +897,13 @@ public function showGrading($gradeable_id, $who_id = '', $from = "", $to = null,
...
 
             // Get the graded gradeable for the $from user
-            $from_graded_gradeable = $this->tryGetGradedGradeable($gradeable, $from, false);
+            $from_graded_gradeable = null;
+            if ($peer && !$team) {
+                $from_graded_gradeable = $this->tryGetGradedGradeable($gradeable, $this->core->getQueries()->getUserFromAnon($from)[$from], false);
+            }
+            else {
+                $from_graded_gradeable = $this->tryGetGradedGradeable($gradeable, $from, false);
+            }
```
[the remark this answers is not in the document]
Comment (jchulton) [reply site/app/controllers/grading/ElectronicGraderController.php]: I tried this, but errored out

[the remark this answers is not in the document]
Comment (emaicus) [reply site/app/controllers/grading/ElectronicGraderController.php]: Had my logic backward. Try:
```
$my_id = $peer && !$team ? $this->core->getQueries()->getUserFromAnon($from)[$from] : $from;
$from_graded_gradeable = $this->tryGetGradedGradeable($gradeable, $my_id, false);
```

**Pull request merged**